enterprise/server/testutil/enterprise_testenv/enterprise_testenv.go

```diff
@@ -60,3 +63,20 @@ func GetCustomTestEnv(t *testing.T, opts *Options) *testenv.TestEnv {
 	env.SetUserDB(userDB)
 	return env
 }
+
+func AddClientIdentity(t *testing.T, te *testenv.TestEnv, client string) {
+	flags.Set(t, "app.client_identity.client", client)
+	key, err := random.RandomString(16)
+	require.NoError(t, err)
+	flags.Set(t, "app.client_identity.key", string(key))
+	require.NoError(t, err)
+	err = clientidentity.Register(te)
+	require.NoError(t, err)
+	require.NotNil(t, te.GetClientIdentityService())
+}
+
```

Comment:
`RunTestCaches` doesn't have anything enterprise specific it looks like, can that be part of `testcache` package?

Reply:
Good call. Moved to `testcache.SetupAndRun`.